Update to v7.8.3

bwingconda opened the pull request:
suitesparse 7.8.3

**Destination channel:** defaults

### Links

- [PKG-7078](https://anaconda.atlassian.net/browse/PKG-7078)
- [Upstream repository](https://github.com/DrTimothyAldenDavis/SuiteSparse/tree/v7.8.3)
- [Upstream changelog/diff](https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/v7.8.3/ChangeLog)
- Relevant dependency PRs:
  - AnacondaRecipes/intel_repack-feedstock#26
  - AnacondaRecipes/mkl-service-feedstock#6

### Explanation of changes:

- Use new `gmp` and `mpfr` on Windows
  - Let `run_exports` populate runtime requirements
- Update `tbb` to what `mkl` uses
- Update to latest version since `7.0.1` no longer builds on Windows

[PKG-7078]: https://anaconda.atlassian.net/browse/PKG-7078?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

bwingconda commented:
Windows build succeeds but fails on the signing workflow. This does not affect the build post-PR merge.

JamesRobertsonGames commented on recipe/bld.bat:
```diff
@@ -3,26 +3,20 @@
-          -DCMAKE_BUILD_TYPE=Release ^
-          -DCMAKE_PREFIX_PATH=%PREFIX% ^
-          -DCMAKE_INSTALL_PREFIX="%LIBRARY_PREFIX%" ^
+          -DBUILD_SHARED_LIBS=ON ^
+          -DBUILD_STATIC_LIBS=ON ^
+          -DCMAKE_BUILD_TYPE:STRING=Release ^
+          -DCMAKE_INSTALL_PREFIX:PATH="%LIBRARY_PREFIX%" ^
+          -DCMAKE_PREFIX_PATH:PATH="%LIBRARY_PREFIX%" ^
```
Why are the options for windows this and for other platforms it is

`CMAKE_OPTIONS="-DCMAKE_INSTALL_PREFIX=${PREFIX} -DCMAKE_INSTALL_LIBDIR=lib -DENABLE_CUDA=0"`

Should these be brought in line with one another or is the difference okay?

bwingconda replied:
I believe the difference is okay because with the newest version it auto-detects CUDA and the pathing should work out correctly with or without the libdir set (unless we're avoiding `lib64` here maybe). In any case, you're right and they should be synced up to avoid confusion later.

JamesRobertsonGames reviewed (commented):
Minor comment

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -15,30 +15,31 @@ build:
     - {{ pin_subpackage("suitesparse") }}
   missing_dso_whitelist:
     - '*/libomp.dylib'    # [osx]
+    - '*/libiomp5md.dll'  # [win]
```
`intel-openmp` should have it in `Library/bin/libiomp5md.dll` or in `bin/libiomp5md.dll`. Is there something wrong with our `intel-openmp 2025?`

skupr-anaconda replied:
So, it actually this:
```
  ERROR (suitesparse,Library/bin/cholmod.dll): Needed DSO Library/bin/libiomp5md.dll found in ['https://staging.continuum.io/prefect/fs/intel_repack-feedstock/pr26/7ae728e/win-64::intel-openmp==2025.0.0=haa95532_1164']

  ERROR (suitesparse,Library/bin/cholmod.dll): .. but ['https://staging.continuum.io/prefect/fs/intel_repack-feedstock/pr26/7ae728e/win-64::intel-openmp==2025.0.0=haa95532_1164'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
```

bwingconda replied:
Yes, currently debugging. I think it's just that it should have been in the runtime requirements rather than the whitelist.

bwingconda replied:
Looks like it's good to go now.

bwingconda committed:
Update to v7.8.3

Remove extra whitelist item

Ensure intel-openmp is in run requirements

Removed abs.yaml
